x/milestone/abci/abci.go

```diff
@@ -67,23 +55,25 @@ func GenMilestoneProposition(ctx sdk.Context, milestoneKeeper *keeper.Keeper, co
 		StartBlockNumber: propStartBlock,
 	}
 
-	SetPendingMilestoneProposition(milestoneProp)
-
 	return milestoneProp, nil
 }
 
-func GetMajorityMilestoneProposition(ctx sdk.Context, validatorSet stakeTypes.ValidatorSet, extVoteInfo []abciTypes.ExtendedVoteInfo, logger log.Logger) (*sidetxs.MilestoneProposition, []byte, string, error) {
+func GetMajorityMilestoneProposition(ctx sdk.Context, validatorSet stakeTypes.ValidatorSet, extVoteInfo []abciTypes.ExtendedVoteInfo, logger log.Logger, lastEndBlock *uint64) (*sidetxs.MilestoneProposition, []byte, string, error) {
 	ac := address.HexCodec{}
 
-	hashToProp := make(map[string]*sidetxs.MilestoneProposition)
-	hashToVotingPower := make(map[string]int64)
-	hashToAggregatedProposersHash := make(map[string][]byte)
-	hashVoters := make(map[string][]string)
+	// Track voting power per block number
```

Comment:
CometBFT doesn't allow duplicate votes but in side tx votes aggregation logic I see that we have check for duplicate votes, so for consistency I think we can add it here too - https://github.com/0xPolygon/heimdall-v2/blob/develop/app/vote_ext_utils.go#L263

Reply:
Good point, will add this back.